addons/base_automation/models/base_automation.py

```diff
@@ -1006,22 +1033,61 @@ def get_record_dt(record):
                             last_run,
                             compute_leaves=True,
                         )
-                    is_process_to_run = past_last_run[calendar.id] <= fields.Datetime.to_datetime(record_dt) < past_now[calendar.id]
+                    return past_last_run[calendar.id] <= record_dt < past_now[calendar.id]
+
+                records = records.filtered_domain(calendar_filter)
```

Comment:
`batch_target_size` isn't take in account for this case? 
`batch_target_size` adds a lot of complexity. My guess, that's not worth it right now, we can still add it later if it really matters.